apps/arena/lib/arena/game_updater.ex

```diff
@@ -151,6 +155,17 @@ defmodule Arena.GameUpdater do
     {:noreply, state}
   end
 
+  def handle_cast(:toggle_bots, state) do
+    encoded_msg =
+      GameEvent.encode(%GameEvent{
+        event: {:toggle_bots, %ToggleBots{}}
+      })
+
+    PubSub.broadcast(Arena.PubSub, state.game_state.game_id, {:toggle_bots, encoded_msg})
```

Comment:
Why is this message broadcasted to the clients? The only difference turning on or off the bots makes to them is received through the `GameState` message, they don't handle this message in any way.

Client PR for reference: [[GH-1831] Add toggle bots button](https://github.com/lambdaclass/champions_of_mirra/pull/1840)

Reply:
The bots are clients as well. Said so, the bots ARE handling that message to change its state.
You can check it in `apps/bot_manager/lib/game_socket_handler.ex`

Also, remember the bots are deployed in a different server than where the backend runs (arena), so we need to send the message somehow (I use this broadcast for that)

Reply:
Great, that makes sense!